File: src/nl/rubensten/texifyidea/structure/LatexSubParagraphPresentation.java
package nl.rubensten.texifyidea.structure;

import com.intellij.navigation.ItemPresentation;
import nl.rubensten.texifyidea.TexifyIcons;
import nl.rubensten.texifyidea.psi.LatexCommands;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public class LatexSubParagraphPresentation implements ItemPresentation {

    private final String subParagraphName;

    public LatexSubParagraphPresentation(LatexCommands subParagraphCommand) {
        if (!subParagraphCommand.getCommandToken().getText().equals("\\subparagraph")) {
            throw new IllegalArgumentException("command is no \\subparagraph-command");
        }

        this.subParagraphName = subParagraphCommand.getRequiredParameters().get(0);
    }

    @Nullable
    @Override
    public String getPresentableText() {
        return subParagraphName;
    }

    @Nullable
    @Override
    public String getLocationString() {
        return null;
    }

    @Nullable
    @Override
    public Icon getIcon(boolean b) {
        return TexifyIcons.DOT_SUBPARAGRAPH;
    }
}

File: src/nl/hannahsten/texifyidea/structure/latex/LatexSubParagraphPresentation.kt
package nl.hannahsten.texifyidea.structure.latex

import nl.hannahsten.texifyidea.TexifyIcons
import nl.hannahsten.texifyidea.psi.LatexCommands
import nl.hannahsten.texifyidea.structure.EditableHintPresentation

/**
 * @author Hannah Schellekens
 */
class LatexSubParagraphPresentation(subParagraphCommand: LatexCommands) : EditableHintPresentation {

    private val subParagraphName = subParagraphCommand.requiredParameterText(0) ?: "Unknown subparagraph"
    private var hint = ""

    override fun getPresentableText() = subParagraphName

    override fun getLocationString() = hint

    override fun getIcon(b: Boolean) = TexifyIcons.DOT_SUBPARAGRAPH

    override fun setHint(hint: String) {
        this.hint = hint
    }
}